xdsl/backend/riscv/register_allocation.py

```diff
@@ -26,6 +26,12 @@ def gather_allocated(func: riscv_func.FuncOp) -> set[RISCVRegisterType]:
         if not isinstance(op, RISCVAsmOperation):
             continue
 
+        if isinstance(op, riscv_func.CallOp):
```

Comment:
Probably worth putting a command here as to why this is done for whoever is going to implement call ABI to find and remove this.